From 0888b61cf3cfbc256e19573e3bf5bfdfd39df1db Mon Sep 17 00:00:00 2001
From: Jordan Williams <jordan@jwillikers.com>
Date: Mon, 22 Apr 2024 09:28:04 -0500
Subject: [PATCH 08/21] meson: Use has_type instead of sizeof to detect
 cpu_set_t type

Signed-off-by: Jordan Williams <jordan@jwillikers.com>
(cherry picked from commit 1b443aa6ddb60f7a8d232429444d57ede200790c)
---
 meson.build | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index 9d87d6e0a0..e8ab2bb2b5 100644
--- a/meson.build
+++ b/meson.build
@@ -451,8 +451,7 @@ endforeach
 
 have = cc.has_header('sched.h')
 conf.set10('HAVE_DECL_CPU_ALLOC', have)
-# We get -1 if the size cannot be determined
-have_cpu_set_t = cc.sizeof('cpu_set_t', prefix : '#define _GNU_SOURCE\n#include <sched.h>') > 0
+have_cpu_set_t = cc.has_type('cpu_set_t', args : '-D_GNU_SOURCE', prefix : '#include <sched.h>')
 conf.set('HAVE_CPU_SET_T', have_cpu_set_t ? 1 : false)
 
 have = cc.has_header_symbol('unistd.h', 'environ', args : '-D_GNU_SOURCE')
-- 
2.44.0

